obal/data/modules/check_koji_build.py

```diff
@@ -23,23 +23,35 @@ def main():
     package = module.params['package']
     koji_executable = module.params['koji_executable']
 
+    command = ['buildinfo', nevr]
     try:
-        if tag:
-            command = ['latest-build', '--quiet', tag, package]
-            build = koji(command, koji_executable)
-            build = build.split(' ')[0]
+        output = koji(command, koji_executable)
+    except KojiCommandError as error:
+        module.fail_json(changed=False, msg=error.message, command=error.command)
 
-            module.exit_json(changed=False, tagged_version=build, exists=(nevr == build))
-        else:
-            command = ['buildinfo', nevr]
-            output = koji(command, koji_executable)
-            exists = 'No such build' not in output
+    exists = "BUILD: %s" % (nevr) in output
 
-            module.exit_json(changed=False, exists=exists)
+    if tag:
+        exists_for_tag = False
 
-    except KojiCommandError as error:
-        module.fail_json(changed=False, msg=error.message, command=error.command)
+        for line in output.split("\n"):
+            if line.startswith("Tags:"):
+                exists_for_tag = tag in line
+
+        if not exists_for_tag:
+            command = ['latest-build', '--quiet', tag, package]
 
+            try:
+                build = koji(command, koji_executable)
+            except KojiCommandError as error:
+                module.fail_json(changed=False, msg=error.message, command=error.command)
+
+            build = build.split(' ')[0]
+            module.exit_json(changed=False, exists=exists, tagged_version=build, exists_for_tag=False)
+        else:
+            module.exit_json(changed=False, exists=exists, tagged_version=build, exists_for_tag=True)
```

Comment:
there is no `build` in the `else` branch defined.
you could use `nevr` here, as you checked that it's the tagged version?